### PR TITLE
kube-aws 0.9.1

### PR DESCRIPTION
**Pull request by ArchiFleKs**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by ArchiFleKs** [comment]
Sorry about all this, it is my first time, it should be good now :/

**Comment by dunn** [comment]
A few more things flagged by the CI bot: https://bot.brew.sh/job/Homebrew%20Core/14430/#showFailuresLink

**Comment by ArchiFleKs** [comment]
Hi, i'll try to fix that, I'm having some trouble with HEAD and devel and some changes upstream

**Comment by ArchiFleKs** [comment]
Should be all good now :)

File: Formula/kube-aws.rb [hunk 1]
```diff
@@ -14,44 +14,72 @@ class KubeAws < Formula
     sha256 "6c5d8d12c01d94918ef68bf4d487aa1b6f577bed3aa64145cd58192fdb8aa948" => :yosemite
   end
 
+  devel do
+    url "https://github.com/coreos/kube-aws/archive/v0.9.3-rc.5.tar.gz"
+    sha256 "0c7d3e6d40adc137d6ff9202400c72b832db4e85ed60abd6dfdb17a9ae301a03"
+    version "0.9.3-rc.5"
+  end
```
Comment (MikeMcQuaid) [inline Formula/kube-aws.rb]: Add newline below here.

File: Formula/kube-aws.rb [hunk 2]
```diff
@@ -14,44 +14,72 @@ class KubeAws < Formula
...
-    kube_aws_dir = "#{gopath_kube_aws}/src/github.com/coreos/coreos-kubernetes/multi-node/aws"
+    kube_aws_dir = "#{gopath_kube_aws}/src/github.com/coreos/kube-aws"
 
     mkdir_p gopath_vendor
     mkdir_p File.dirname(kube_aws_dir)
 
-    ln_s buildpath/"multi-node/aws/vendor", "#{gopath_vendor}/src"
-    ln_s buildpath/"multi-node/aws", kube_aws_dir
+    ln_s buildpath/"vendor", "#{gopath_vendor}/src"
+    ln_s buildpath/".", kube_aws_dir
```
Comment (MikeMcQuaid) [inline Formula/kube-aws.rb]: `ln_s buildpath, kube_aws_dir`

File: Formula/kube-aws.rb [hunk 3]
```diff
@@ -14,44 +14,72 @@ class KubeAws < Formula
...
+        system "go", "generate", "./config"
+        system "go", "generate", "./nodepool/config"
+        system "go", "build", "-ldflags",
+               "-X github.com/coreos/kube-aws/cluster.VERSION=#{version}",
+               "-a", "-tags", "netgo", "-installsuffix", "netgo",
+               "-o", bin/"kube-aws", "./"
+      end
+    else
+      cd kube_aws_dir do
+        system "go", "generate", "./config"
```
Comment (MikeMcQuaid) [inline Formula/kube-aws.rb]: The `cd` and `go generate` can be put outside the `if/`else`

File: Formula/kube-aws.rb [hunk 4]
```diff
@@ -14,44 +14,72 @@ class KubeAws < Formula
...
+      system "#{bin}/kube-aws"
+      cluster = { "clusterName" => "test-cluster", "externalDNSName" => "dns",
+                  "keyName" => "key", "region" => "west",
+                  "availabilityZone" => "zone", "kmsKeyArn" => "arn" }
+      system "#{bin}/kube-aws", "init", "--cluster-name", "test-cluster",
+             "--external-dns-name", "dns", "--region", "west",
+             "--availability-zone", "zone", "--key-name", "key",
+             "--kms-key-arn", "arn"
+      cluster_yaml = YAML.load_file("cluster.yaml")
+      assert_equal cluster, cluster_yaml
```
Comment (MikeMcQuaid) [inline Formula/kube-aws.rb]: Similarly a bunch of the above can be put outside the `if/else`. Can you also share the parts of the hash where possible? Thanks!

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by ArchiFleKs** [comment]
@MikeMcQuaid : Thanks, I made the changes. What do you mean by parts of the hash ?

File: Formula/kube-aws.rb [hunk 5]
```diff
@@ -14,43 +14,61 @@ class KubeAws < Formula
...
     ENV["GOPATH"] = "#{gopath_vendor}:#{gopath_kube_aws}"
 
     cd kube_aws_dir do
-      system "go", "generate", "./pkg/config"
-      system "go", "build", "-ldflags",
-             "-X github.com/coreos/coreos-kubernetes/multi-node/aws/pkg/cluster.VERSION=#{version}",
-             "-a", "-tags", "netgo", "-installsuffix", "netgo",
-             "-o", bin/"kube-aws", "./cmd/kube-aws"
+      system "go", "generate", "./config"
+      if build.devel? || build.head?
```
Comment (MikeMcQuaid) [inline Formula/kube-aws.rb]: Flip the `if`/`else` and do `if build.stable?` ... `else`. Don't do `unless` ...`else`. Thanks!

File: Formula/kube-aws.rb [hunk 6]
```diff
@@ -14,43 +14,61 @@ class KubeAws < Formula
...
+      end
     end
   end
 
   test do
     system "#{bin}/kube-aws"
-
-    cluster = { "clusterName" => "test-cluster", "externalDNSName" => "dns",
-                "keyName" => "key", "region" => "west",
-                "availabilityZone" => "zone", "kmsKeyArn" => "arn" }
```
Comment (MikeMcQuaid) [inline Formula/kube-aws.rb]:
```ruby
cluster = { "clusterName" => "test-cluster", "externalDNSName" => "dns",
                  "keyName" => "key", "region" => "west",
                  "availabilityZone" => "zone", "kmsKeyArn" => "arn" }
cluster["controller"] = nil unless build.stable?
```

and then you can remove the `if`/`else`

File: Formula/kube-aws.rb [hunk 7]
```diff
@@ -14,43 +14,61 @@ class KubeAws < Formula
...
+                  "keyName" => "key", "region" => "west",
+                  "availabilityZone" => "zone", "kmsKeyArn" => "arn" }
+    end
     system "#{bin}/kube-aws", "init", "--cluster-name", "test-cluster",
            "--external-dns-name", "dns", "--region", "west",
            "--availability-zone", "zone", "--key-name", "key",
            "--kms-key-arn", "arn"
     cluster_yaml = YAML.load_file("cluster.yaml")
     assert_equal cluster, cluster_yaml
-
```
Comment (MikeMcQuaid) [inline Formula/kube-aws.rb]: Leave this empty line.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by MikeMcQuaid** [comment]
@BrewTestBot test this please

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by nijikon** [comment]
You are awesome, thanks for your contribution! :tada:

**Pull request closed without merging**